tests/e2e/support/deploy-to-netlify.ts

```diff
@@ -0,0 +1,107 @@
+ * @param fixtureName name of the folder inside the fixtures folder
+ */
+export const deployFixture = async (fixtureName: string): Promise<Fixture> => {
+  const isolatedFixtureRoot = await prepareFixture(fixtureName)
+  const result = await deploySite(isolatedFixtureRoot)
+  return result
+}
+
+/**
+ * Copy a fixture site to an isolated temp directory, prepare its dependencies, link it against any
+ * applicable local packages, and make it deployable to Netlify.
+ */
+const prepareFixture = async (fixtureName: string): Promise<string> => {
+  const isolatedFixtureRoot = await mkdtemp(join(tmpdir(), 'netlify-remix-compute-e2e-'))
+  console.log(`📂 Copying fixture '${fixtureName}' to '${isolatedFixtureRoot}'...`)
+
+  const src = fileURLToPath(new URL(`../fixtures/${fixtureName}`, import.meta.url))
+  const files = await fg.glob('**/*', {
+    ignore: ['node_modules'],
+    dot: true,
+    cwd: src,
+  })
+
+  const limit = pLimit(Math.max(2, cpus().length))
+  await Promise.all(
+    files.map((file) =>
+      limit(async () => {
+        await mkdir(join(isolatedFixtureRoot, dirname(file)), { recursive: true })
+        await copyFile(join(src, file), join(isolatedFixtureRoot, file))
+      }),
+    ),
+  )
+
+  await prepareDeps(isolatedFixtureRoot, resolve('.', 'packages'))
+
+  await execaCommand('git init', { cwd: isolatedFixtureRoot })
+
+  return isolatedFixtureRoot
+}
+
+const packages = [
+  { name: '@netlify/remix-adapter', dirName: 'remix-adapter' },
+  { name: '@netlify/remix-edge-adapter', dirName: 'remix-edge-adapter' },
+  { name: '@netlify-remix-runtime', dirName: 'remix-runtime' },
+]
+
+/**
+ * Inject the current revision of this repo's packages into the fixture.
+ *
+ * We can't use a simpler approach like a monorepo workspace or `npm link` because the fixture site
+ * needs to be self-contained to be deployable to Netlify (i.e. it can't have symlinks).
+ */
+const prepareDeps = async (cwd: string, packagesAbsoluteDir: string): Promise<void> => {
+  await execaCommand('pnpm install', { cwd })
+  const { dependencies, devDependencies } = JSON.parse(await readFile(`${cwd}/package.json`, 'utf-8'))
+  for (const pkg of packages) {
+    if (pkg.name in dependencies || pkg.name in devDependencies) {
+      const isDevDep = pkg.name in devDependencies
+      console.log(`📦 Injecting ${pkg.name} ${isDevDep ? 'dev ' : ''}dependency...`)
+      const { stdout } = await execaCommand(`npm pack --json --ignore-scripts --pack-destination ${cwd}`, {
+        cwd: join(packagesAbsoluteDir, pkg.dirName),
+      })
+      const [{ filename }] = JSON.parse(stdout)
+      await execaCommand(`pnpm add ${isDevDep ? '--save-dev' : ''} file:${join(cwd, filename)}`, { cwd })
+    }
+  }
+}
+
+const deploySite = async (isolatedFixtureRoot: string): Promise<Fixture> => {
+  console.log(`🚀 Building and deploying site...`)
+
+  const outputFile = 'deploy-output.txt'
+  const cmd = `pnpm exec ntl deploy --build --site ${SITE_ID}`
```

Comment:
I'm not certain on this, but you might not need `pnpm exec` here. The `netlify-cli` should be installed globally for users at this point, and I believe it's pre-installed in our organizations github actions.

Reply:
Yes, all GH actions have ntl pre-installed

Reply:
Just note that `netlify-cli` version that comes with default action runners won't always be `latest` and in case we would need updates we will have to either wait or install latest manually ourselves

Also unclear if that's actually the case but `ubuntu-24.04` image doesn't seem to come with `netlify-cli` installed ( `ubuntu-22.04` aka current `ubuntu-latest` does have it listed):
 - https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md (has `Netlify CLI 17.22.1`)
 - https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md (doesn't have netlify cli listed)

So maybe to be safe - we do want to install CLI ourselves to shield ourselves from future changes to images? Especially if we want to replicate setup created here for other adapters?